src/vs/workbench/contrib/debug/browser/debugSessionPicker.ts

```diff
@@ -0,0 +1,119 @@
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { IDebugService, IDebugSession, REPL_VIEW_ID } from 'vs/workbench/contrib/debug/common/debug';
+import { IQuickInputService, IQuickPickSeparator } from 'vs/platform/quickinput/common/quickInput';
+
+import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
+import { IPickerDebugItem } from 'vs/workbench/contrib/debug/common/loadedScriptsPicker';
+import { IViewsService } from 'vs/workbench/common/views';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+
+
+export async function showDebugSessionMenu(accessor: ServicesAccessor, selectAndStartID: string) {
+	const quickInputService = accessor.get(IQuickInputService);
+	const debugService = accessor.get(IDebugService);
+	const viewsService = accessor.get(IViewsService);
+	const commandService = accessor.get(ICommandService);
+
+	const localDisposableStore = new DisposableStore();
+	const quickPick = quickInputService.createQuickPick<IPickerDebugItem>();
+	localDisposableStore.add(quickPick);
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = _getPicks(quickPick.value, selectAndStartID, debugService, viewsService, commandService);
+
+	localDisposableStore.add(quickPick.onDidChangeValue(async () => {
+		quickPick.items = _getPicks(quickPick.value, selectAndStartID, debugService, viewsService, commandService);
+	}));
+	localDisposableStore.add(quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		quickPick.hide();
+		localDisposableStore.dispose();
+	}));
+	quickPick.show();
+}
+
+function _getPicks(filter: string, selectAndStartID: string, debugService: IDebugService, viewsService: IViewsService, commandService: ICommandService): Array<IPickerDebugItem | IQuickPickSeparator> {
+	const debugConsolePicks: Array<IPickerDebugItem | IQuickPickSeparator> = [];
+	const headerSessions: IDebugSession[] = [];
+
+	const sessions = debugService.getModel().getSessions(false);
+
+	sessions.forEach((session) => {
+		if (session.compact && session.parentSession) {
+			headerSessions.push(session.parentSession);
+		}
+	});
+
+	sessions.forEach((session) => {
+		const isHeader = headerSessions.includes(session);
+		if (!session.parentSession) {
+			debugConsolePicks.push({ type: 'separator', label: isHeader ? session.name : undefined });
+		}
+
+		if (!isHeader) {
+			const pick = _createPick(session, filter, debugService, viewsService, commandService);
+			if (pick) {
+				debugConsolePicks.push(pick);
+			}
+		}
+	});
+
+	if (debugConsolePicks.length > 0) {
+		debugConsolePicks.push({ type: 'separator' });
+	}
+
+	const createTerminalLabel = nls.localize("workbench.action.debug.startDebug", "Start a New Debug Session");
```

Comment:
Use single quotes for localized string IDs. Thought we would have a lint rule for that, guess not